src/plugins/controls/public/control_types/range_slider/range_slider_embeddable.tsx

```diff
@@ -0,0 +1,312 @@
+  public deferEmbeddableLoad = true;
+
+  private subscriptions: Subscription = new Subscription();
+  private node?: HTMLElement;
+
+  // Controls services
+  private dataService: ControlsDataService;
+  private dataViewsService: ControlsDataViewsService;
+
+  // Internal data fetching state for this input control.
+  private dataView?: DataView;
+  private field?: DataViewField;
+
+  // State to be passed down to component
+  private componentState: RangeSliderComponentState;
+  private componentStateSubject$ = new BehaviorSubject<RangeSliderComponentState>({
+    min: '',
+    max: '',
+    loading: true,
+    fieldFormatter: (value: string) => value,
+  });
+
+  constructor(input: RangeSliderEmbeddableInput, output: ControlOutput, parent?: IContainer) {
+    super(input, output, parent); // get filters for initial output...
+
+    // Destructure controls services
+    ({ data: this.dataService, dataViews: this.dataViewsService } = pluginServices.getServices());
+
+    this.componentState = {
+      min: '',
+      max: '',
+      loading: true,
+      fieldFormatter: (value: string) => value,
+    };
+    this.updateComponentState(this.componentState);
+
+    this.initialize();
+  }
+
+  private initialize = async () => {
+    const initialValue = this.getInput().value;
+    if (initialValue) {
+      await this.fetchMinMax();
+    }
+    this.setInitializationFinished();
+    this.setupSubscriptions();
+  };
+
+  private setupSubscriptions = () => {
+    const dataFetchPipe = this.getInput$().pipe(
+      map((newInput) => ({
+        lastReloadRequestTime: newInput.lastReloadRequestTime,
+        dataViewId: newInput.dataViewId,
+        fieldName: newInput.fieldName,
+        timeRange: newInput.timeRange,
+        filters: newInput.filters,
+        query: newInput.query,
+      })),
+      distinctUntilChanged(diffDataFetchProps)
+    );
+
+    // fetch available min/max when input changes or when search string has changed
+    this.subscriptions.add(dataFetchPipe.subscribe(this.fetchMinMax));
+
+    // build filters when value change
+    this.subscriptions.add(
+      this.getInput$()
+        .pipe(
+          debounceTime(400),
+          distinctUntilChanged((a, b) => isEqual(a.value, b.value)),
+          skip(1) // skip the first input update because initial filters will be built by initialize.
+        )
+        .subscribe(() => this.fetchMinMax())
+    );
+  };
+
+  private getCurrentDataViewAndField = async (): Promise<{
+    dataView: DataView;
+    field: DataViewField;
+  }> => {
+    const { dataViewId, fieldName } = this.getInput();
+    if (!this.dataView || this.dataView.id !== dataViewId) {
+      this.dataView = await this.dataViewsService.get(dataViewId);
+      if (this.dataView === undefined) {
+        this.onFatalError(
+          new Error(RangeSliderStrings.errors.getDataViewNotFoundError(dataViewId))
+        );
+      }
+      this.updateOutput({ dataViews: [this.dataView] });
```

Comment:
Since the control group hierarchical chaining system is listening to _any_ output changes Outputting the data views every time you get them could cause lower performance due to the diffing functions having to run more than usual. 

I see that you're batching the `updateOutput` in your getMinMax function, so this line could potentially just be removed?